provider/provider.go

```diff
@@ -544,38 +540,16 @@ func hashContext(dockerContextPath string, dockerfile string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("error hashing dockerfile %q: %w", dockerfile, err)
 	}
-	// for each file in the Docker build context, create a hash of its content
-	err = filepath.WalkDir(dockerContextPath, func(path string, d os.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
-		// The relative path is used in checking against .dockerignore and what is used in computing the
-		// name of the file hashed.
-		relPath, err := filepath.Rel(dockerContextPath, path)
+	err = fsutil.Walk(context.Background(), dockerContextPath, &fsutil.WalkOpt{
+		ExcludePatterns: ignorePatterns,
+	}, func(relPath string, d fs.FileInfo, err error) error {
```

Comment:
I'd like to disambiguate these vars a bit - relPath gets me thinking about relative dockerfiles! 
```suggestion
	}, func(filePath string, fileInfo fs.FileInfo, err error) error {
```